src/Core/Commander/Providers/IoDriver_JSON.js

```diff
@@ -19,7 +19,21 @@ define('Core/Commander/Providers/IoDriver_JSON', ['Core/Commander/Providers/IoDr
 
     IoDriver_JSON.prototype.read = function(url) {
         return fetch(url).then(function(response) {
-            return response.json();
+            return response.json().then(
+                function(response) {
+                    if(response.status >= 200 && response.status < 300)
+                        return response;
+                    else{
+                        var error = new Error(response.statusText);
+                        error.response = response;
+                        throw error;
+                    }
+                }).then(
+                function(data){
+                    return data;
+                }).catch(function(error) {
+                    console.log('request failed', error);
```

Comment:
Also, note that this means that the method returns a _fulfilled_ `Promise` (of `undefined`), so it would be complicated for callers of this method to handle errors.

I strongly believe that the error should not be caught here, and we should leave error handling to the "end" of the chain as much as possible (`MobileMappingLayer` currently).

Note that when a promise has been _rejected_, all the `then()` are skipped; only the next `catch()` (or `onRejected` callback passed as a second argument tp a `then()`) will be called, so all intermediary steps can assume things went well in their `then()` (this is assuming this change is not merged, where each `then()` would have to check whether the argument it receives is `undefined`)